lte/gateway/python/integ_tests/s1aptests/test_attach_with_mme_restart_after_request.py

```diff
@@ -0,0 +1,123 @@
+"""
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import time
+import unittest
+
+import orc8r.protos.metricsd_pb2 as metricsd
+import s1ap_types
+import s1ap_wrapper
+from s1ap_utils import MagmadUtil
+
+class TestAttachWithMmeRestartAfterRequest(unittest.TestCase):
+
+    def setUp(self):
+        self._s1ap_wrapper = s1ap_wrapper.TestWrapper(
+            stateless_mode=MagmadUtil.stateless_cmds.ENABLE)
+
+    def tearDown(self):
+        self._s1ap_wrapper.cleanup()
+
+    def restart_mme_and_wait(self):
+        print("************************* Restarting mme service on gateway")
+        self._s1ap_wrapper.magmad_util.restart_services(["mme"])
+        print("************************* Waiting for mme to restart. 20 sec")
+        time.sleep(20)
+
+    def test_attach_with_mme_restart_after_request(self):
+        num_ues = 1
+        self._s1ap_wrapper.configUEDevice(num_ues)
+        print("************************* Sending Attach Request for ue-id : 1")
+        attach_req = s1ap_types.ueAttachRequest_t()
+        attach_req.ue_Id = 1
+        sec_ctxt = s1ap_types.TFW_CREATE_NEW_SECURITY_CONTEXT
+        id_type = s1ap_types.TFW_MID_TYPE_IMSI
+        eps_type = s1ap_types.TFW_EPS_ATTACH_TYPE_EPS_ATTACH
+        attach_req.mIdType = id_type
+        attach_req.epsAttachType = eps_type
+        attach_req.useOldSecCtxt = sec_ctxt
+
+        self._s1ap_wrapper._s1_util.issue_cmd(
+            s1ap_types.tfwCmd.UE_ATTACH_REQUEST, attach_req
+        )
+
+        response = self._s1ap_wrapper.s1_util.get_response()
+        self.assertEqual(
+            response.msg_type, s1ap_types.tfwCmd.UE_AUTH_REQ_IND.value
+        )
+        print("************************* Received UE_AUTH_REQ_IND")
+
+        self.restart_mme_and_wait()
```

Comment:
isn't this better be issued after the attach request command is sent or rename the test to `test_attach_with_mme_restart_after_auth_request`?

Reply:
Problem is the MME would have sent the response back by the time it got the restart signal. Not easy to synchronize that